crates/schema/src/error.rs

```diff
@@ -0,0 +1,181 @@
+use spacetimedb_data_structures::error_stream::ErrorStream;
+use spacetimedb_lib::db::{
+    error::{self, DefType},
+    raw_def::IndexType,
+};
+use spacetimedb_primitives::{ColId, ColList};
+use spacetimedb_sats::{typespace::TypeRefError, AlgebraicType, AlgebraicTypeRef};
+use std::borrow::Cow;
+use std::fmt;
+
+use crate::identifier::Identifier;
+
+pub type ValidationErrors = ErrorStream<ValidationError>;
+
+#[derive(thiserror::Error, Debug, PartialOrd, Ord, PartialEq, Eq)]
+#[non_exhaustive]
+pub enum ValidationError {
+    #[error("name `{name}` is used for multiple entities")]
+    DuplicateName { name: Identifier }, // use an Identifier because this is checked for validated names.
```

Comment:
add: most other variants of this enum store un-validated strings, since we want to support giving an error message even if names aren't valid.